eos/scattering/ee-to-ccbar.cc

```diff
@@ -0,0 +1,778 @@
+
+            // Channel properties
+            const double Nf = 2 * K->_channels[Channels(channel)]->_l_orbital + 1;
+            const double rhof = std::abs(K->_channels[Channels(channel)]->rho(intermediate_result->s));
+
+            // Get T-matrix[ee, channel]
+            const complex<double> T1f = intermediate_result->tmatrix_row_0[Channels(channel)];
+
+            return sqrt(GeVtonb / intermediate_result->s * Nf * rhof)* T1f;
+        }
+
+        complex<double> T_eetochannel(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            // Get T-matrix[ee, channel]
+            const complex<double> T1f = intermediate_result->tmatrix_row_0[Channels(channel)];
+
+            return T1f;
+        }
+
+        complex<double> T2_eetochannel(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            // Get T-matrix[ee, channel] on the second RS
+            const complex<double> T1f = intermediate_result->tmatrix2_row_0[Channels(channel)];
+
+            return T1f;
+        }
+
+        double sigma_eetochannel(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            return norm(amplitude_eetochannel(intermediate_result, channel));
+        }
+
+        // Widths
+        double res_partial_width(const Resonances & resonance, const Channels & channel)
+        {
+            return K->partial_width(resonance, channel);
+        }
+
+        double res_total_width(const Resonances & resonance)
+        {
+            return K->width(resonance);
+        }
+
+        // Ratios
+        double R(const IntermediateResult * intermediate_result)
+        {
+            double total_xsec = 0.0;
+
+            for (unsigned i = 1; i < EEToCCBar::nchannels; i++)
+            {
+                total_xsec += sigma_eetochannel(intermediate_result, Channels(i));
+            }
+
+            return total_xsec / sigma_eetomumu(abs(intermediate_result->E)) + Rconstant; // Add constant term
+        }
+
+        // Spectral function
+        double spectral_function(const double E, const Resonances & resonance)
+        {
+            return K->spectral_function(resonance, E*E);
+        }
+    };
+
+    EEToCCBar::EEToCCBar(const Parameters & parameters, const Options & options) :
+        PrivateImplementationPattern<EEToCCBar>(new Implementation<EEToCCBar>(parameters, options, *this))
+    {
+    }
+
+    EEToCCBar::~EEToCCBar()
+    {
+    }
+
+    const std::vector<OptionSpecification>
+    Implementation<EEToCCBar>::options
+    {
+        {"assume_isospin", { "true", "false" }, "false"},
+    };
+
+    const EEToCCBar::IntermediateResult *
+    EEToCCBar::prepare(const double & E) const
+    {
+        return _imp->prepare(E);
+    }
+
+    const EEToCCBar::IntermediateResult *
+    EEToCCBar::prepare_complex(const double & reE, const double & imE) const
```

Comment:
Coding style: this looks very much like camel case. Can we use ``re_E`` instead, to use at least snake case with accidentally upper case letters :smile:

Reviewed until here.